### PR TITLE
Deprecate default value for `trustServerCertificate ` being `true`

### PR DESCRIPTION
**Pull request by arthurschreiber**
Defaulting `trustServerCertificate` to `true` is a potential security issue, so we should switch to it being set to false, and maybe even update our documentation to guide users to use the correct `cryptoCredentialsDetails` values to verify server certificates instead of relying on `trustServerCertificate`.

References: https://github.com/tediousjs/tedious/issues/983

### CODE REVIEW

File: src/connection.ts [hunk 1]
```diff
@@ -912,6 +912,8 @@ class Connection extends EventEmitter {
         }
 
         this.config.options.trustServerCertificate = config.options.trustServerCertificate;
+      } else {
+        deprecate('The default value for `config.options.trustServerCertificate` will change from `true` to `false` in the next major version of `tedious`. Set the value to `true` or `false` explicitly to silence this message.');
```
Comment (MichaelSun90) [inline src/connection.ts]: Hi @arthurschreiber. I just had a question. Is this change only a message for users to get aware of the change ahead? The actual default value change for line 537 will be in another PR?

Comment (arthurschreiber) [reply src/connection.ts]: Yes exactly. If we change the default, a lot of applications that rely on `trustServerCertificate` defaulting to `true` will break, so I'd like to do that change together with a major version bump.

By emitting a value when `trustServerCertificate` is not set (and not emitting the message if a value is provided), we can nudge users to set a value explicitly. If we then change the default, they will be unaffected (because they provided an explicit value).

**Comment by IanChokS** [comment]
@arthurschreiber Will the breaking change be merged along side the dropping Node.js 6/8 support?

**Comment by arthurschreiber** [comment]
Yeah, I'd just group all pending breaking changes (that are ready) together, so we only bump the major version once.

**Comment by arthurschreiber** [comment]
> Yeah, I'd just group all pending breaking changes (that are ready) together, so we only bump the major version once.

Just a note that this is not a hard requirement. I'm also fine bumping the major version multiple times, if that means we can move a bit quicker and don't get hold up. 🤷‍♀
